server/server.go

```diff
@@ -18,63 +18,81 @@ package server
 
 import (
 	"fmt"
-	"sync"
-	"time"
 
 	"metacontroller.io/controller/common"
 
 	"metacontroller.io/controller/decorator"
 	"metacontroller.io/options"
 
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
 	"metacontroller.io/apis/metacontroller/v1alpha1"
 	mcclientset "metacontroller.io/client/generated/clientset/internalclientset"
 	"metacontroller.io/controller/composite"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 
 	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
 )
 
-type controller interface {
-	Start()
-	Stop()
-}
+// New returns a new controller manager and a function which can be used
+// to release resources after the manager is stopped.
+func New(configuration options.Configuration) (controllerruntime.Manager, func(), error) {
+	mgr, err := controllerruntime.NewManager(configuration.RestConfig, manager.Options{
```

Comment:
@fpetkovski can we :
1. move `mcClient` declaration before
2. move `controllerContext` declaration before
3. refactor `ControllerContext` in common package, to have `broadcaster` public
4. Set `EventBroadcaster` instance in `manager.Options` to the same instance
5. (Optional) if it would be  possible, replace and remove the one created in the `ControllerContext`

main motivation is that we have options to limit `spam level` of broadcaster in input flags and would be good to have all instances using the same limit

Reply:
if removal of mode from `ControllerContext` will be to big/to complicated leave it, it can be removed later

Reply:
Alright, 1-4 should be done. 

I did not quite understand point 5. What specifically would you suggest we replace and remove in `ControllerContext`?